plugins/module_utils/network/common/network_template.py

```diff
@@ -0,0 +1,94 @@
+import re
+from copy import deepcopy
+
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    Template,
+    dict_merge,
+)
+
+
+class NetworkTemplate(object):
```

Comment:
can we move the netcommon changes to diff PR, as it can be referred separately in future

Reply:
You are right. The netcommon PR (https://github.com/ansible-collections/ansible.netcommon/pull/21) is blocked by a NETCONF issue now.